qiskit/quantum_info/states/statevector.py

```diff
@@ -359,6 +360,29 @@ def reverse_qargs(self):
         ret._op_shape = self._op_shape.reverse()
         return ret
 
+    def expectation_value_pauli(self, pauli):
+        """Compute the expectation value of a Pauli.
+
+            Args:
+                pauli (Pauli): a Pauli operator to evaluate expval of.
+
+            Returns:
+                complex: the expectation value.
+            """
+        # pylint: disable=no-name-in-module
+        from .cython.exp_value import expval_pauli_no_x, expval_pauli_with_x
+        x_mask = sum([2**k for k in range(len(pauli)) if pauli.x[k]])
```

Comment:
Should use numpy here to avoid slow python loops over arrays:
```python
n_pauli = len(pauli)
x_mask = np.dot(1 << np.arange(n_pauli), pauli.x)
z_mask = np.dot(1 << np.arange(n_pauli), pauli.z)
phase = (-1j) ** np.sum(pauli.x & pauli.z)
```